### PR TITLE
Improve notice description #392

### LINKED ISSUE #392: [REQUEST] Ease tracking of faulty GTFS rows
**Is your feature request related to a problem? Please describe.**
It is sometime difficult to track down faulty GTFS entities with the information given in the notice description.

**Describe the solution you'd like**
Improve said description, add element to improve tracking of faulty GTFS entitie

**Additional context**
External contribution, question 2: https://github.com/MobilityData/gtfs-validator/issues/390

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #392 
**Summary:**

This PR provides support to improve notice descriptions

**Expected behavior:** 

To be updated 

Please make sure these boxes are checked before submitting your pull request - thanks!
- [x] remove line separator in notice descriptions
- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- ~[ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")~
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Comment by lionel-nj** [comment]
@maximearmstrong  note that notice memory consumption tests fail on CircleCI (for 1000 and 10000 notices) but they pass locally and in Github Actions.

**Comment by lionel-nj** [comment]
I actually remember that we did not want these tests to be hardware dependant, but the experience shows different behavior in Circle CI and GitHub actions. I'll add this subject to the next technical meeting agenda: do you think tha we should run test with different thresholds depending on whether CI is GitHub Actions or Circle CI? @maximearmstrong  @fabrice-v

**Comment by maximearmstrong** [comment]
@lionel-nj I don't think we should run different tests (thresholds) depending on which CI runs the code. This being said, what makes tests fail on Circle CI? Is it the `SAFETY_BUFFER_FACTOR` that is not high enough or is it failing before assertion? If the problem comes from the safety threshold, I suggest increasing it. If not, we should investigate what fails. If we consider keeping both GitHub CI and Circle CI in the future, one option would be to disable that test on Circle CI as it could be linked to its memory consumption. CC @fabrice-v

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/FastTravelBetweenStopsNotice.java [hunk 1]
```diff
@@ -25,12 +25,13 @@
...
-                                        final float speedkmh,
+                                        final float speedKmh,
                                         final List<Integer> stopSequenceList) {
         super("stop_times.txt", E_046,
                 "Fast travel between stops",
-                "Fast travel detected in trip:`" + tripId + "` over stop sequence:"
-                        + stopSequenceList + ". Calculated speed: " + speedkmh + "kmh",
+                String.format("Fast travel detected in trip: `%s` over stop sequence: `%s`." +
+                                " Calculated speed: `%s` kmh.",
+                        tripId, stopSequenceList, speedKmh),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/FastTravelBetweenStopsNotice.java:34]: Nothing really, but in any other file, you put one variable per line (for clarity I assume). Maybe just do the same here to maintain the same scheme.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/DuplicatedEntityNotice.java [hunk 2]
```diff
@@ -76,12 +84,14 @@ public DuplicatedEntityNotice(final String filename, final String fieldName,
...
-                "Entity must be unique in file: `" + filename + "` found other entity with same value for " +
-                        "fields: " +
-                        "`" + compositeKeyFirstPart + "`: " + compositeKeyFirstValue + "`" + "--" +
-                        "`" + compositeKeySecondPart + "`: " + compositeKeySecondValue + "`" + "--" +
-                        "`" + compositeKeyThirdPart + "`: " + compositeKeyThirdValue + "`" + "--" +
-                        "`" + compositeKeyFourthPart + "`: " + compositeKeyFourthValue + "`.", null);
+                String.format("Entity must be unique in file: `%s`. Found other entity with same value for " +
+                                "fields: (`%s`=`%s` ; `%s`=`%s` ; `%s`=`%s` ; `%s`=`%s`)",
+                        filename,
+                        compositeKeyFirstPart, compositeKeyFirstValue,
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/DuplicatedEntityNotice.java:90]: One variable per line would be better for readability.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/FeedInfoExpiresInLessThan7DaysNotice.java [hunk 3]
```diff
@@ -35,13 +35,16 @@ public FeedInfoExpiresInLessThan7DaysNotice(final String fileName,
...
-                        "`" + compositeKeySecondPart + "`: `" + compositeKeySecondValue + "` -- " +
-                        "`" + compositeKeyThirdPart + "`: `" + compositeKeyThirdValue + "`.",
+                String.format("GTFS dataset should be valid for at least the next 7 days. Current date is: `%s`. " +
+                                "Feed expires: `%s` in field `%s` of file `%s` for entity with composite id: " +
+                                "(`%s`=`%s` ; `%s`=`%s` ; `%s`=`%s`)",
+                        currentDateAsString,
+                        feedEndDateAsString,
+                        fieldName,
+                        fileName,
+                        compositeKeyFirstPart, compositeKeyFirstValue,
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/FeedInfoExpiresInLessThan7DaysNotice.java:45]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
     public MissingHeaderNotice(final String filename, final String missingHeaderName) {
         super(filename, E_001,
                 "Missing required field",
-                String.format("`%s` is missing required field: `%s`.", filename, missingHeaderName),
+                String.format("GTFS file `%s` is missing required field: `%s`.", filename, missingHeaderName),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java:29]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingRequiredFileNotice.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
     public MissingRequiredFileNotice(String filename) {
         super(filename, E_003,
                 "Missing required file",
-                "File `" + filename + "` is required.",
+                String.format("GTFS file `%s` is required.", filename),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingRequiredFileNotice.java:28]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/ExtraFileFoundNotice.java [hunk 6]
```diff
@@ -27,12 +27,12 @@ public ExtraFileFoundNotice(String filename) {
         super(filename,
                 W_004,
                 "File name not defined by the GTFS specification",
-                "Extra file `" + filename + "` found in archive",
+                String.format("Extra file `%s` found in archive", filename),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/ExtraFileFoundNotice.java:30]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/InputZipContainsFolderNotice.java [hunk 7]
```diff
@@ -27,7 +27,7 @@ public InputZipContainsFolderNotice(final String filename, final String folderNa
         super(filename,
                 W_001,
                 "Non empty folder",
-                "File: `" + filename + "`contains folder named: `" + folderName + "`",
+                String.format("File: `%s` contains folder named: `%s`", filename, folderName),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/InputZipContainsFolderNotice.java:30]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/MissingRouteLongNameNotice.java [hunk 8]
```diff
@@ -26,7 +26,8 @@
     public MissingRouteLongNameNotice(final String filename, final String entityId) {
         super(filename, W_007,
                 "Empty `route_long_name`",
-                "Empty `route_long_name` for route with `route_id`:`" + entityId + "` in file:`" + filename + "`",
+                String.format("Empty `route_long_name` for route with `route_id`: `%s` in file: `%s`",
+                        entityId, filename),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/MissingRouteLongNameNotice.java:30]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/MissingRouteShortNameNotice.java [hunk 9]
```diff
@@ -26,7 +26,8 @@
     public MissingRouteShortNameNotice(final String filename, final String entityId) {
         super(filename, W_006,
                 "Empty `route_short_name`",
-                "Empty `route_short_name` for route with id:" + entityId + "` in file:" + filename + "`",
+                String.format("Empty `route_short_name` for route with `route_id`: `%s` in file: `%s`",
+                        entityId, filename),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/MissingRouteShortNameNotice.java:30]: Same as above.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/RouteLongNameContainsShortNameNotice.java [hunk 10]
```diff
@@ -26,8 +26,8 @@
     public RouteLongNameContainsShortNameNotice(final String filename, final String entityId) {
         super(filename, W_008,
                 "`route_long_name` contains `route_short_name`",
-                "`route_long_name` contains `route_short_name`:`" + entityId + "` in file:`" + filename +
-                        "`. `route_long_name` and should not contain the `route_short_name`.",
+                String.format("`route_long_name` contains `route_short_name` in file `%s`. `route_long_name`" +
+                                " and should not contain the `route_short_name`.", filename),
```
Comment (maximearmstrong) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/warning/RouteLongNameContainsShortNameNotice.java:30]: Same as above.

**Review by maximearmstrong — CHANGES_REQUESTED** [review]
Minor request changes. Most of your `String.format()` displays one variable per line, which makes it very easy to read the code. In some cases, there are many variables per line, making it harder to read. Other than that, looks good!

**Comment by lionel-nj** [comment]
> @lionel-nj I don't think we should run different tests (thresholds) depending on which CI runs the code. This being said, what makes tests fail on Circle CI? Is it the SAFETY_BUFFER_FACTOR that is not high enough or is it failing before assertion? If the problem comes from the safety threshold, I suggest increasing it. If not, we should investigate what fails. If we consider keeping both GitHub CI and Circle CI in the future, one option would be to disable that test on Circle CI as it could be linked to its memory consumption. CC @fabrice-v

Actually these failures are totally random so I could not determine the source of the the failure. As you can see:
- 1 job that fails: https://app.circleci.com/pipelines/github/MobilityData/gtfs-validator/393/workflows/8ecd5f22-3a14-427f-8977-4c5e92959a66
- successful re-run of the same job: https://app.circleci.com/pipelines/github/MobilityData/gtfs-validator/393/workflows/0c684c9b-cab2-43cc-b451-b3dc976ea801

It is also to note that even bumping the threshold did not stop this job from failing.

**Comment by lionel-nj** [comment]
To answer your question, from what I witnessed the tests fail at the assertion steps, which leads me to think that the thresholds was not sufficient. However, the same tests pass locally and on GitHub and I doubt that we should continue to extend these thresholds, otherwise we would not be testing thoroughly

**Pull request closed without merging**